app/forms/sufia/forms/work_form.rb

```diff
@@ -2,8 +2,15 @@ module Sufia::Forms
   class WorkForm < CurationConcerns::Forms::WorkForm
     delegate :depositor, :on_behalf_of, :permissions, to: :model
 
+    attr_reader :agreement_accepted
+
     self.terms += [:collection_ids]
 
+    def initialize(model, current_ability)
+      @agreement_accepted = !model.new_record?
```

Comment:
@jcoyne as you suggested I changed the code to use `model.new_record?` (instead of model.file_sets.count > 1). This should be pretty fast, right?